web/client-api/src/main/java/io/deephaven/web/client/api/JsPartitionedTable.java

```diff
@@ -283,15 +271,37 @@ public Column[] getColumns() {
      *
      * @return Promise of a Table
      */
-    @JsMethod
+    @Deprecated
     public Promise<JsTable> getKeyTable() {
-        return keys.copy();
+        if (keys == null) {
+            keys = connection.newState((c, state, metadata) -> {
+                SelectOrUpdateRequest view = new SelectOrUpdateRequest();
+                view.setSourceId(baseTable.state().getHandle().makeTableReference());
+                view.setResultId(state.getHandle().makeTicket());
+                view.setColumnSpecsList(descriptor.getKeyColumnNamesList());
+                connection.tableServiceClient().view(view, metadata, c::apply);
+            }, "view only keys columns")
```

Comment:
```suggestion
            }, "view only key columns")
```

I think that reads better?